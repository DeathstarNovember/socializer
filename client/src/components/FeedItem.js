import React from "react";
import { Card } from "react-bootstrap";
import { Link } from "react-router-dom";
import Gravatar from "react-gravatar";
import Timeago from "react-timeago";
import renderIf from "render-if";

const FeedItem = ({ feedType, item }) => {
  return (
    <Card className="mb-4">
      <Card.Header className="d-flex justify-content-between align-items-center">
        <div className="d-flex align-items-center">
          <Gravatar
            md5={item.user.gravatarMd5}
            className="rounded-circle"
            size={30}
          />
          <div className="ml-2">
            <div className="h6 mb-0">{item.user.name}</div>
          </div>
        </div>
        <div className="text-muted h7">
          <Timeago date={`${item.insertedAt}Z`} />
        </div>
      </Card.Header>
      <Card.Body>
        <p className="card-text">{item.body}</p>
      </Card.Body>
      {renderIf(feedType === "post")(
        <Card.Footer className="d-flex">
          <Link to={`/posts/${item.id}`} className="ml-auto">
            Discussion â‰«
          </Link>
        </Card.Footer>,
      )}
    </Card>
  );
};

export default FeedItem;
