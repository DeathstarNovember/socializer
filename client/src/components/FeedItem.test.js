import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import tk from "timekeeper";
import FeedItem from "./FeedItem";

describe("FeedItem", () => {
  const item = {
    id: 123,
    body: "Some thoughts",
    insertedAt: "2019-04-17T16:00:00",
    user: {
      name: "John Smith",
      gravatarMd5: "abcdefg",
    },
  };

  beforeEach(() => {
    tk.freeze("2019-04-18");
  });

  afterEach(() => {
    tk.reset();
  });

  it("renders correctly for post", () => {
    const { container } = render(
      <MemoryRouter>
        <FeedItem feedType="post" item={item} />,
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly for comment", () => {
    const { container } = render(
      <MemoryRouter>
        <FeedItem feedType="comment" item={item} />,
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
