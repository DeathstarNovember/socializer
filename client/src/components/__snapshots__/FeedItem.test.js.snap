// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem renders correctly for comment 1`] = `
<div
  class="mb-4 card"
>
  <div
    class="d-flex justify-content-between align-items-center card-header"
  >
    <div
      class="d-flex align-items-center"
    >
      <img
        alt="Gravatar for undefined"
        class="react-gravatar rounded-circle"
        height="30"
        src="//www.gravatar.com/avatar/abcdefg?d=retro&r=g&s=30"
        srcset="//www.gravatar.com/avatar/abcdefg?d=retro&r=g&s=60 2x"
        width="30"
      />
      <div
        class="ml-2"
      >
        <div
          class="h6 mb-0"
        >
          John Smith
        </div>
      </div>
    </div>
    <div
      class="text-muted h7"
    >
      <time
        datetime="2019-04-17T16:00:00.000Z"
        title="2019-04-17T16:00:00Z"
      >
        8 hours ago
      </time>
    </div>
  </div>
  <div
    class="card-body"
  >
    <p
      class="card-text"
    >
      Some thoughts
    </p>
  </div>
</div>
`;

exports[`FeedItem renders correctly for post 1`] = `
<div
  class="mb-4 card"
>
  <div
    class="d-flex justify-content-between align-items-center card-header"
  >
    <div
      class="d-flex align-items-center"
    >
      <img
        alt="Gravatar for undefined"
        class="react-gravatar rounded-circle"
        height="30"
        src="//www.gravatar.com/avatar/abcdefg?d=retro&r=g&s=30"
        srcset="//www.gravatar.com/avatar/abcdefg?d=retro&r=g&s=60 2x"
        width="30"
      />
      <div
        class="ml-2"
      >
        <div
          class="h6 mb-0"
        >
          John Smith
        </div>
      </div>
    </div>
    <div
      class="text-muted h7"
    >
      <time
        datetime="2019-04-17T16:00:00.000Z"
        title="2019-04-17T16:00:00Z"
      >
        8 hours ago
      </time>
    </div>
  </div>
  <div
    class="card-body"
  >
    <p
      class="card-text"
    >
      Some thoughts
    </p>
  </div>
  <div
    class="d-flex card-footer"
  >
    <a
      class="ml-auto"
      href="/posts/123"
    >
      Discussion â‰«
    </a>
  </div>
</div>
`;
